httpd/http_server.c

```diff
@@ -0,0 +1,333 @@
+        freez(query_unescaped);
+    }
+
+    web_client_api_request_v1(*host, &w, path_unescaped);
+    freez(path_unescaped);
+
+    h2o_iovec_t body = buffer_to_h2o_iovec(w.response.data);
+
+    // we move msg body to req->pool managed memory as it has to
+    // live until whole response has been encrypted and sent
+    // when req is finished memory will be freed with the pool
+    void *managed = h2o_mem_alloc_shared(&req->pool, body.len, NULL);
+    memcpy(managed, body.base, body.len);
+    body.base = managed;
+
+    req->res.status = 200;
+    req->res.reason = "OK";
+    if (w.response.data->content_type == CT_APPLICATION_JSON)
+        h2o_add_header(&req->pool, &req->res.headers, H2O_TOKEN_CONTENT_TYPE, NULL, CONTENT_JSON_UTF8);
+    else
+        h2o_add_header(&req->pool, &req->res.headers, H2O_TOKEN_CONTENT_TYPE, NULL, CONTENT_TEXT_UTF8);
+    h2o_start_response(req, &generator);
+    h2o_send(req, &body, 1, H2O_SEND_STATE_FINAL);
+
+    buffer_free(w.response.data);
+    buffer_free(w.response.header);
+
+    return 0;
+}
+
+static int netdata_uberhandler(h2o_handler_t *self, h2o_req_t *req)
+{
+    RRDHOST *host = localhost;
+
+    int ret = _netdata_uberhandler(self, req, &host);
+
+    char host_uuid_str[GUID_LEN + 1];
+    uuid_unparse_lower(host->host_uuid, host_uuid_str);
+
+    if (!ret) {
+        log_access("HTTPD OK method: " PRINTF_H2O_IOVEC_FMT
+                   ", path: " PRINTF_H2O_IOVEC_FMT
+                   ", as host: %s"
+                   ", response: %d",
+                   PRINTF_H2O_IOVEC(&req->method),
+                   PRINTF_H2O_IOVEC(&req->input.path),
+                   host == localhost ? "localhost" : host_uuid_str,
+                   req->res.status);
+    } else {
+        log_access("HTTPD %d"
+                   " method: " PRINTF_H2O_IOVEC_FMT
+                   ", path: " PRINTF_H2O_IOVEC_FMT
+                   ", forwarding to file handler as path: " PRINTF_H2O_IOVEC_FMT,
+                   ret,
+                   PRINTF_H2O_IOVEC(&req->method),
+                   PRINTF_H2O_IOVEC(&req->input.path),
+                   PRINTF_H2O_IOVEC(&req->path));
+    }
+
+    return ret;
+}
+
+static int hdl_netdata_conf(h2o_handler_t *self, h2o_req_t *req)
+{
+    if (!h2o_memis(req->method.base, req->method.len, H2O_STRLIT("GET")))
+        return -1;
+
+    BUFFER *buf = buffer_create(NBUF_INITIAL_SIZE_RESP, NULL);
+    config_generate(buf, 0);
+
+    void *managed = h2o_mem_alloc_shared(&req->pool, buf->len, NULL);
+    memcpy(managed, buf->buffer, buf->len);
+
+    req->res.status = 200;
+    req->res.reason = "OK";
+    h2o_add_header(&req->pool, &req->res.headers, H2O_TOKEN_CONTENT_TYPE, NULL, CONTENT_TEXT_UTF8);
+    h2o_send_inline(req, managed, buf->len);
+    buffer_free(buf);
+
+    return 0;
+}
+
+void *httpd_main(void *ptr) {
+    h2o_pathconf_t *pathconf;
+    h2o_hostconf_t *hostconf;
+
+    const char *bind_addr = config_get(HTTPD_CONFIG_SECTION, "bind to", "127.0.0.1");
```

Comment:
Can the user give different IPs and ports like we do with our web server?  What about our current permissions? Are we going to discard them completely?

Reply:
do they make sense?
old webserver can be called by DASHBOARD, ACLK etc. but new webserver currently serve only DASHBOARD

Reply:
point of this PR is to be minimal functional first step to build upon, not a final product

Reply:
Probably only due backward compatibility, but considering this is a complete new webserver we also could discard. I think this needs a better internal discussion ( ping @stelfrag and @cakrit ).

Reply:
As this is a secondary web server that is only enabled on demand, there's nothing wrong with missing features.

Reply:
Thank you @cakrit !